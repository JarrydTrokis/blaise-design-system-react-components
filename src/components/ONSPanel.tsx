import React from "react";

export interface Props {
    children: any,
    status?: "success" | "error" | "info" | "warn",
    spacious?: boolean,
    id?: string,
    hidden?: boolean,
    testID?: string
    bigIcon?: boolean
}

export const ONSPanel = (props: Props) => {
    const className = "panel panel--" + (props.status ? props.status : "info") + " panel--no-title " + (props.spacious ? "panel--spacious" : "") + " u-mt-m";
    return (
        <div data-testid={props.testID} id={props.id} className={className} hidden={props.hidden}>
            {
                props.status === "success" &&
                <span className="panel__icon">
                <svg className={`svg-icon ${props.bigIcon === true ? "svg-icon--xl" : ""}`} viewBox="0 0 13 10" xmlns="http://www.w3.org/2000/svg">
                <path
                    d="M14.35,3.9l-.71-.71a.5.5,0,0,0-.71,0h0L5.79,10.34,3.07,7.61a.51.51,0,0,0-.71,0l-.71.71a.51.51,0,0,0,0,.71l3.78,3.78a.5.5,0,0,0,.71,0h0L14.35,4.6A.5.5,0,0,0,14.35,3.9Z"
                    transform="translate(-1.51 -3.04)"/>
                </svg>
                </span>
            }
            {
                props.status === "warn" &&
                <span className="panel__icon" aria-hidden="true">!</span>
            }
            <div className={`panel__body ${props.bigIcon === true ? "svg-icon-margin--xl" : ""}`}>
                {props.children}
            </div>
        </div>
    );
};