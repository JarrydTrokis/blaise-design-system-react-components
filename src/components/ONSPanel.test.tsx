import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import {ONSPanel} from "./ONSPanel";
import Adapter from "enzyme-adapter-react-16";

describe("ONS Panel Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    const panelProps = {
        children: <p>Succceeesssss</p>,
    };

    const statusPanelProps = {
        children: <p>Statusssss</p>,
        status: "success",
        // spacious: false
    };

    const spaciousPanelProps = {
        children: <p>Spaciooouuusssss</p>,
        status: "error",
        spacious: true,
        id: "spacious"
    }

    const bigIconStatusPanelProps = {
        children: <p>Statusssss</p>,
        status: "success",
        bigIcon: true
        // spacious: false
    };
    
    function wrapper (render: any, props: any) {
        return render(
            <ONSPanel   {...props}
                        >{props.children}</ONSPanel>)}

    it("matches Snapshot", () => {
        expect(wrapper(shallow, panelProps)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, panelProps).exists()).toEqual(true));


    it("should render the correct children", () => {
        expect(wrapper(shallow, panelProps).find("div.panel__body").getElement().props.children).toEqual(panelProps.children)
    })

    it("should render the correct status", () => {
        expect(wrapper(shallow, statusPanelProps).find("div.panel").hasClass('panel--success')).toEqual(true)
    })
    
    it('displays a spacious panel button', () => {
        expect(wrapper(shallow, spaciousPanelProps).find('div.panel').hasClass('panel--spacious')).toEqual(true)
    })

    it("should render the big success tick", () => {
        expect(wrapper(shallow, bigIconStatusPanelProps).find("svg.svg-icon").hasClass('svg-icon--xl')).toEqual(true)
        expect(wrapper(shallow, bigIconStatusPanelProps).find("div.panel__body").hasClass('svg-icon-margin--xl')).toEqual(true)
    })

    it("matches Snapshot a big success icon", () => {
        expect(wrapper(shallow, bigIconStatusPanelProps)).toMatchSnapshot();
    })
})