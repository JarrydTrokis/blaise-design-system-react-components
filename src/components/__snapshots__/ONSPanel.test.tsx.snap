// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Panel Test matches Snapshot 1`] = `ShallowWrapper {}`;

exports[`ONS Panel Test matches Snapshot a big success icon 1`] = `ShallowWrapper {}`;
